tracee-rules/input.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"bufio"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+)
+
+func setupStdinSource() (chan types.Event, error) {
+	res := make(chan types.Event)
+	scanner := bufio.NewScanner(os.Stdin)
+	go func() {
+		for scanner.Scan() {
+			res <- types.TraceeEvent{
+				ArgsNum: 1,
+				Args: []types.TraceeEventArgument{
+					{
+						Name:  "pathname",
+						Value: scanner.Text(),
+					},
+				},
+			}
+		}
+		close(res)
+	}()
+	return res, nil
+}
+
+func setupTraceeSource(traceeFilePath string) (chan types.Event, error) {
+	f, err := os.Open(traceeFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("invalid file: %s", traceeFilePath)
+	}
+	dec := gob.NewDecoder(f)
+	res := make(chan types.Event)
+	go func() {
+		for {
+			var event types.TraceeEvent
+			err := dec.Decode(&event)
+			if err != nil {
+				if err == io.EOF {
+					break
```

Comment:
I couldn't test the streaming mode because of local issues that I have in my dev env, but looking at the gob test that we have in tracee, it breaks on EOF and it does handle streaming: https://github.com/aquasecurity/tracee/blob/main/test/gob/gob.go#L43
Are you sure that this is the case?

Reply:
If this is the case, then it's a problem because the goroutine will never exit and also the input channel will never be closed

Reply:
Have a look at the current sigengine code. It handles the streaming case as well.
I understand that this may be trickier with this design that makes use of channels.
As we control tracee code (luck us!) maybe we can add an event that signals the end of the stream when tracee exits.
We should probably also have a done channel here to exit gracefully

Reply:
I saw it, but it didn't make sense to me so I didn't use it, instead I based it on the gob test which I knew was working in streaming. I have just tested a "streaming" case with tracee and break on EOF and it seems fine. can you share what issue you face exactly?

Reply:
I'm unable to make tracee-rules work on my end

Reply:
why not? how can I help?

I'm testing the following on my end which looks fine:
```
sudo ./tracee -o gob | ./tracee-rules --tracee-file /proc/self/fd/0
```

Reply:
[sudo] password for yaniv: 
panic: interface conversion: interface {} is int32, not string

goroutine 21 [running]:
plugin/unnamed-ee3d7f777123a9839007ea47a11c8f48fcbdbed2.(*counter).OnEvent(0x7f593c3e39b0, 0x96fb20, 0xc00024ec60, 0x0, 0x0)
	/home/yaniv/src/tracee/tracee-rules/signatures/golang/example.go:54 +0x3e5
github.com/aquasecurity/tracee/tracee-rules/engine.Engine.Start.func1(0x7f593c369700, 0x7f593c3e39b0, 0xc000018300)
	/home/yaniv/src/tracee/tracee-rules/engine/engine.go:62 +0x93
created by github.com/aquasecurity/tracee/tracee-rules/engine.Engine.Start
	/home/yaniv/src/tracee/tracee-rules/engine/engine.go:60 +0x116

Reply:
recompile the plugin

Reply:
the example interface here is very naive.. it assumes the first arg is a string.. try to run tracee with `-e security_bprm_check` for example. I'll add a check to the sig as well

Reply:
pushed a fix

Reply:
I'm not getting a match for the signature even thogh I changed "yo" to "/usr/bin/ls" (and recompiled)

Reply:
resolved based on https://github.com/aquasecurity/tracee/pull/379